src/SQLStore/PropertyTableInfoFetcher.php

```diff
@@ -0,0 +1,197 @@
+<?php
+
+namespace SMW\SQLStore;
+
+use SMW\DataTypeRegistry;
+use SMW\ApplicationFactory;
+use SMW\DIProperty;
+use SMWDataItem as DataItem;
+
+/**
+ * @license GNU GPL v2+
+ * @since 2.2
+ *
+ * @author mwjames
+ */
+class PropertyTableInfoFetcher {
+
+	/**
+	 * Array for keeping property table table data, indexed by table id.
+	 * Access this only by calling getPropertyTables().
+	 *
+	 * @var TableDefinition[]
+	 */
+	private static $propertyTableDefinitions = null;
```

Comment:
We need the cache here because of performance, this information is accessed several hundreds times per request and the current way the` SQLStore` is using the information can't be circumvented.

> I think it's harmful to make such a cache global.

I don't want the `SQLStore` to make this sort of caching therefore `PropertyTableInfoFetcher` is the solely keeper.

First step is to isolate and keep status quo but allow for a possibility adjustments at a later stage.